cpp/src/arrow/record_batch_test.cc

```diff
@@ -64,44 +65,98 @@ class TestRecordBatch : public ::testing::Test {};
 TEST_F(TestRecordBatch, Equals) {
   const int length = 10;
 
+  auto f0 = field("f0", int32());
+  auto f1 = field("f1", uint8());
+  auto f2 = field("f2", int16());
+
+  auto schema0_f0_f1_f2 = schema({f0, f1, f2});
+  auto schema1_f0_f1_f2 = schema({f0, f1, f2});
+  auto schema2_f0_f1 = schema({f0, f1});
```

Comment:
Can we use intention instead of content for name?

```suggestion
  auto schema = schema({f0, f1, f2});
  auto schema_same = schema({f0, f1, f2});
  auto schema_different = schema({f0, f1});
```